tests/test_cache.py

```diff
@@ -0,0 +1,83 @@
+import filecmp
+import pytest
+import tempfile
+import os
+
+from pathlib import Path
+from unittest import mock
+
+from dnadoh.cache import FileCache
+
+def _create_file_in_remote(test_fname, cacher):
+    target_path = Path(cacher.cache_dir, test_fname)
+    source_path = Path(cacher.remote_url, test_fname)
+    # remove the file if exists 
+    if os.path.exists(target_path):
```

Comment:
https://pypi.org/project/pyfakefs/ is your best friend here - you want to avoid littering the hard drive with actual files if you can.

Reply:
implemented pyfakefs in my tests and don't think it broke anything...